### PR TITLE
[chore] Clarify config removal in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by axw**
#### Description

Clarify that deprecated config options are expected to remain for at least 6 months after deprecation.

"MUST NOT" makes it clear for readers that this is something that definitely should not be done, vs. something that's discretionary.

See also  https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33776#issuecomment-2217491855

### CODE REVIEW

**Review by crobert-1 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
cc @open-telemetry/collector-approvers will merge by Wednesday EU morning unless somebody blocks by then

**Review by songy23 — APPROVED** [review]

**Pull request merged**
